### ISSUE TITLE
Quickstart Vagrant tutorial coming out soon - looking for reviewers

### ISSUE THREAD

**Issue by Swader** [sulu-io/sulu-standard#552]
The quickstart Vagrant tutorial (with optimizations) is coming out this week or the next.
I'd appreciate it if someone could take a look, give me some feedback if you feel like something needs changing, etc.

This is just the quick start, a full exploratory tutorial series is coming in the second half of November / early December.

If you'd like to lend a hand with reviewing the post, please ping me here and I'll add you to our repo where all article drafts go as PRs. You can then drop inline comments there.

**Comment by wachterjohannes** [comment]
@Swader i have not really experiences with vagrant but perhaps i could give you some inputs (:

**Comment by danrot** [comment]
@Swader I guess you also still have the problem https://github.com/sulu-io/sulu-standard/issues/541, so this would be the perfect opportunity to have a look at both.

@wachterjohannes Do you want to take this one?

**Comment by Swader** [comment]
@danrot #541 is indeed still a problem, but I'll have it tracked down by the end of the day, no doubt.

@wachterjohannes the tutorial is literally a step by step procedure to get everything running - a sequential list of explained commands, so it should be very easy to follow and will be of utmost importance to your future users.

@danrot can I see a changelog for 1.1? Maybe something changed enough to warrant a change in the article.

**Comment by danrot** [comment]
@Swader of course, the changelog is also in the repository: https://github.com/sulu-io/sulu/blob/develop/CHANGELOG.md But it's really a lot... We've summed that a bit up in a [blog article](http://blog.sulu.io/sulu-1-1-0-first-beta-release). The main points are a big redesign of the UI, the security system and making it possible to use the smart content not only for pages.

**Comment by danrot** [comment]
@Swader I have just talked to @wachterjohannes, can you give me access to the article, then I will review it, and try to tackle https://github.com/sulu-io/sulu-standard/issues/541

**Comment by Swader** [comment]
Will do, as soon as it's ready to be reviewed. Adding some more fancy automation to the vagrant moving script - it'll automatically detect the app's name, adjusting the vendor folder's name accordingly, thus supporting multiple sulu installs on a single machine, even with moved vendor folders. When is 1.1 to be released? Maybe I should wait for it before publishing.

**Comment by danrot** [comment]
There will be a release candidate next monday, then it depends on how much critical bugs we encounter, but I guess another 2 or 3 weeks.

**Comment by Swader** [comment]
Ok, vendor moving script is ironed out, now with some added safeguards.
@danrot you should be able to follow instructions in this writeup and get to the point where I'm getting the #541 error within 5 minutes (depending on how fast Composer will finish, and how good of a download speed you'll have downloading the box for the first time).

http://www.sitepoint.com/?p=118527&preview=1&_ppp=5c599b0d88

Made a public preview link, so everyone else is welcome to chime in as well.

**Comment by wachterjohannes** [comment]
@Swader i will take a look at this on the weekend (:

**Comment by danrot** [comment]
* In the "Clone / Install Sulu" I would place the database name before the credentials, since it is also in this order in the post install
* You should run `app/console sulu:build dev` or `app/console sulu:build prod`, that's not clear to me from the description. I would also make more clear that `app/console sulu:build dev` should absolutely not be run on production, since it creates a default user with an easy to guess password.
* The line `~/Code/bin/sulu/vendorfix.sh` doesn't work for me, in ~/Code/bin is only the file `folderfix.sh`

**Comment by Swader** [comment]
> The line ~/Code/bin/sulu/vendorfix.sh doesn't work for me, in ~/Code/bin is only the file folderfix.sh

Fixed. File was missing from repo, forgot to push.

> You should run app/console sulu:build dev or app/console sulu:build prod, that's not clear to me from the description. I would also make more clear that app/console sulu:build dev should absolutely not be run on production, since it creates a default user with an easy to guess password.

Is it "or"? I was under the impression (from the sulu docs) that both need to be run in development, and the `prod` one only in production, no? I have changed the `dev` paragraph to indicate danger of running it in production.

> In the "Clone / Install Sulu" I would place the database name before the credentials, since it is also in this order in the post install

Fixed, thanks.

**Comment by danrot** [comment]
Well, the commands don't hurt if they are run like this. Both contain a set of tasks that must be fulfilled, and prod is a subset of dev. But you are right, our docs are also a bit confusing at this place, we should fix that.

**Comment by Swader** [comment]
:+1: 

Now that the showstopping bugs are out of the way, I'm continuing to polish the post. Any other feedback is very welcome.

**Comment by Swader** [comment]
Article updated. I believe it's ready now.
http://www.sitepoint.com/?p=118527&preview=1&_ppp=5c599b0d88

**Comment by danrot** [comment]
I've seen you still have both versions of the `app/console sulu:build` command in your article. As said, I would change that to only use one of them, this way the same tasks are executed twice.

**Comment by Swader** [comment]
Ahh, I guess I misunderstood *again*. I thought that in dev, you *should* run both, and in prod, only the prod one. Will amend, thanks

**Comment by danrot** [comment]
Now it's right :-) But I realized two other things:

* If you run `app/console sulu:build dev` there is already a role with all available permissions at that time, and the user with the credentials admin/admin, so it is not really required to create another role and user to start working with Sulu (although you might want to do that, what I want to say, is that it is not necessary)
* For changing to dev mode I would rather use the `SYMFONY_ENV` environment variable, and set it via the webserver configuration (`SetEnv` on apache or `fastcgi_param` on nginx).

**Comment by nielsvermaut** [comment]
If you still have a need for the reviewer, I helped with the vagrant environment at work for our Symfony applications, and I'd be happy to help.

**Comment by Swader** [comment]
@nielsvermaut yes please - drop feedback here!
